src/Tickets/Emails/Email_Abstract.php

```diff
@@ -828,4 +828,101 @@ public function get_post_type_data(): array {
 
 		return $data;
 	}
+
+	/**
+	 * Get a successful order array for preview.
+	 * 
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	public function get_preview_order_successful(): array {
```

Comment:
Also, this shouldn't live in abstract. The goal of an abstract class is to be extended, storing common functionality that's applicable to any class that's extending it. If we're using `order` data in one or two emails, then these things should live there. Not in the abstract class.

Reply:
I moved all preview data into its own class called `Preview_Data`. Any of the emails can fetch the data from there.